src/memory/server.ts

```diff
@@ -1,34 +1,162 @@
 import { SDKService } from "@buf/lekkodev_sdk.bufbuild_connect-es/lekko/server/v1beta1/sdk_connect";
-import { ConnectRouter } from "@bufbuild/connect";
+import { ConfigurationService } from '@buf/lekkodev_sdk.bufbuild_connect-es/lekko/client/v1beta1/configuration_service_connect';
+import { Code, ConnectError, ConnectRouter } from "@bufbuild/connect";
 import { connectNodeAdapter } from "@bufbuild/connect-node";
-import { Http2Server, createServer } from 'http2';
-import { Store } from "./store";
+import { NotFoundError } from "./store";
+import * as http from "http";
+import { GetBoolValueResponse, GetFloatValueResponse, GetIntValueResponse, GetJSONValueResponse, GetProtoValueResponse, GetStringValueResponse, Value } from '@buf/lekkodev_sdk.bufbuild_es/lekko/client/v1beta1/configuration_service_pb';
+import { Client, DevClient } from '../types/client';
+import { ClientContext } from '../context/context';
+
+const LOCAL_PATH_HEADER = "localpath";
 
 // Runs a simple vanilla nodejs web server for debugging. 
 // The server exposes the interface defined here: 
 //      https://buf.build/lekkodev/sdk/docs/main:lekko.server.v1beta1
 export class SDKServer {
-    server?: Http2Server;
+    client: Client & DevClient;
+    server?: http.Server;
 
-    constructor(store: Store, port?: number) {
+    constructor(client: Client & DevClient, port?: number) {
+        this.client = client;
         if (!port) {
             return;
         }
-        const routes = (router: ConnectRouter) => 
+        const routes = (router: ConnectRouter) => {
             router.service(SDKService, {
-                async listContents() {
-                    return store.listContents();
+                listContents: async () => {
+                    return this.client.listContents();
+                }
+            });
+            // TODO: Clean up code once interceptor support lands in connect-node
+            router.service(ConfigurationService, {
+                getBoolValue: async (req, context) => {
+                    await this.handleHeaders(context.requestHeader);
+                    try {
+                        await this.handleHeaders(context.requestHeader);
```

Comment:
unless you really need to do it twice :)
```suggestion
                    await this.handleHeaders(context.requestHeader);
                    try {
```

Reply:
Oops, nice catch!